test/e2e_node/device_plugin_test.go

```diff
@@ -589,6 +589,55 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 				framework.Fail(fmt.Sprintf("stale device assignment after pod deletion while kubelet was down allocated=%v error=%v", allocated, err))
 			}
 		})
+
+		ginkgo.It("Can schedule a pod with a restartable init container [NodeAlphaFeature:SidecarContainers]", func(ctx context.Context) {
+			podRECMD := fmt.Sprintf("devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep %s", sleepIntervalWithRestart)
+			pod1 := e2epod.NewPodClient(f).CreateSync(ctx, makeBusyboxPodWithRestartableInitContainer(SampleDeviceResourceName, podRECMD))
+			deviceIDRE := "stub devices: (Dev-[0-9]+)"
+			devID1, err := parseLog(ctx, f, pod1.Name, pod1.Spec.InitContainers[0].Name, deviceIDRE)
+			framework.ExpectNoError(err, "getting logs for pod %q/%q", pod1.Name, pod1.Spec.InitContainers[0].Name)
+			gomega.Expect(devID1).To(gomega.Not(gomega.Equal("")), "pod1's restartable init container requested a device but started successfully without")
+
+			devID2, err := parseLog(ctx, f, pod1.Name, pod1.Spec.Containers[0].Name, deviceIDRE)
+			framework.ExpectNoError(err, "getting logs for pod %q/%q", pod1.Name, pod1.Spec.Containers[0].Name)
+			gomega.Expect(devID2).To(gomega.Not(gomega.Equal("")), "pod1's regular container requested a device but started successfully without")
+
+			gomega.Expect(devID1).NotTo(gomega.Equal(devID2), "pod1's restartable init container and regular container should not share the same device")
+
+			podResources, err := getV1NodeDevices(ctx)
+			framework.ExpectNoError(err)
+
+			framework.Logf("PodResources.PodResources:%+v\n", podResources.PodResources)
+			framework.Logf("len(PodResources.PodResources):%+v", len(podResources.PodResources))
+
+			gomega.Expect(podResources.PodResources).To(gomega.HaveLen(2))
+
+			var resourcesForOurPod *kubeletpodresourcesv1.PodResources
+			for _, res := range podResources.GetPodResources() {
+				if res.Name == pod1.Name {
+					resourcesForOurPod = res
+				}
+			}
+
+			gomega.Expect(resourcesForOurPod).NotTo(gomega.BeNil())
+
+			gomega.Expect(resourcesForOurPod.Name).To(gomega.Equal(pod1.Name))
+			gomega.Expect(resourcesForOurPod.Namespace).To(gomega.Equal(pod1.Namespace))
+
+			// Note that the kubelet does not report resources for init
+			// containers for now.
+			// See (*v1PodResourcesServer).List in pkg/kubelet/apis/podresources/server_v1.go
+			// TBD: Do we have to fix this?
```

Comment:
https://github.com/kubernetes/kubernetes/blob/56cc5e77a10ba156694309d9b6159d4cd42598e1/pkg/kubelet/apis/podresources/server_v1.go#L51-L86

Currently, the kubelet does not report the resources for init containers. Do we need to fix it..??
@SergeyKanzhelev

Reply:
yes, this is an oversight. The reason why the podresource API don't report resources allocated to sidecars is because... the code predates sidecars. The API is meant to report resources allocated to longrunning containers, and at time the initcontainers were all transient.
Feel free to change the test code in a different way which workarounds this bug.
I think fixing podresources API can be a beta item, meaning we should fix this, but is not very urgent. If we want to fix it early to use the API for tests, even better

Reply:
I'll separate it from this PR and create an issue.

Reply:
perfect. Happy to help here.

Reply:
one more question.
Who are the primary consumers of this kubelet API?
(I want to know the impact and urgency of the issue.)

Reply:
The consumers I'm aware of are the SRIOV stack, GPU Monitoring agents and consumers of topology-aware-scheduling. While sidecar feature is alpha and disabled by default I don't think is very urgent. OTOH it should be fixed before move to beta (enabled by default).